examples/vag_cdc.c

```diff
@@ -0,0 +1,260 @@
+
+#define CDC_END_CMD 0x14
+#define CDC_PLAY 0xE4
+#define CDC_STOP 0x10
+#define CDC_NEXT 0xF8
+#define CDC_PREV 0x78
+#define CDC_SEEK_FWD 0xD8
+#define CDC_SEEK_RWD 0x58
+#define CDC_CD1 0x0C
+#define CDC_CD2 0x8C
+#define CDC_CD3 0x4C
+#define CDC_CD4 0xCC
+#define CDC_CD5 0x2C
+#define CDC_CD6 0xAC
+#define CDC_SCAN 0xA0
+#define CDC_SFL 0x60
+#define CDC_PLAY_NORMAL 0x08
+
+#define MODE_PLAY 0xFF
+#define MODE_SHFFL 0x55
+#define MODE_SCAN 0x00
+
+uint16_t captimehi = 0;
+uint16_t captimelo = 0;
+uint8_t capturingstart = 0;
+uint8_t capturingbytes = 0;
+uint32_t cmd = 0;
+uint8_t cmdbit = 0;
+uint8_t newcmd = 0;
+uint8_t shutdownpending = 0;
+
+uint8_t getCommand(uint32_t cmd);
+void shutdown();
+void softreset();
+
+volatile uint8_t cd;
+volatile uint8_t tr;
+volatile uint8_t mode;
+
+ISR(INT0_vect) //remote signals
+{
+	if(RADIO_OUT_IS_HIGH)
+	{
+		if (capturingstart || capturingbytes)
+		{
+			captimelo = TCNT1;
+		}
+		else
+			capturingstart = 1;
+		TCNT1 = 0;
+
+		//eval times
+		if (captimehi > 8300 && captimelo > 3500)
+		{
+			capturingstart = 0;
+			capturingbytes = 1;
+			//uart_puts("startseq found\r\n");
+		}
+		else if(capturingbytes && captimelo > 1500)
+		{
+			//uart_puts("bit 1\r\n");
+			cmd = (cmd<<1) | 0x00000001;
+			cmdbit++;
+		}
+		else if (capturingbytes && captimelo > 500)
+		{
+			//uart_puts("bit 0\r\n");
+			cmd = (cmd<<1);
+			cmdbit++;
+		}
+		else
+		{
+			//uart_puts("nothing found\r\n");
+		}
+		if(cmdbit == 32)
+		{
+			//uart_puts("new cmd\r\n");
+			newcmd = 1;
+			cmdbit = 0;
+			capturingbytes = 0;
+		}
+	}
+	else
+	{
+		captimehi = TCNT1;
+		TCNT1 = 0;
+	}
+}
+
+ISR(INT1_vect) //ft230x cbus1
+{
+	if (PIND & (1<<FT_CBUS1)) //reset radio display to 'PLAY' CD01 TR00
+	{
+		//PORTA &= ~(1<<LED_PWR);
+		cd = 0xBE;
+		tr = 0xFF;
+		mode = 0xFF;
+	}
+	else //usb connect
+	{
+		PORTA |= (1<<LED_PWR);
+		_delay_ms(50);
+		PORTA &= ~(1<<LED_PWR);
+		cd = 0xB0;
+		tr = 0x00;
+	}
+}
+
+uint8_t spi_xmit(uint8_t val) {
+	SPDR = val;
+	while(!(SPSR & (1<<SPIF)));
+	return SPDR;
+}
+
+void send_package(uint8_t c0, uint8_t c1, uint8_t c2, uint8_t c3, uint8_t c4, uint8_t c5, uint8_t c6, uint8_t c7)
+{
+	spi_xmit(c0);
+	_delay_us(874);
+	spi_xmit(c1);
+	_delay_us(874);
+	spi_xmit(c2);
+	_delay_us(874);
+	spi_xmit(c3);
+	_delay_us(874);
+	spi_xmit(c4);
+	_delay_us(874);
+	spi_xmit(c5);
+	_delay_us(874);
+	spi_xmit(c6);
+	_delay_us(874);
+	spi_xmit(c7);
+}
+
+int main(void)
+{
+	cd = 0xBE;
+	tr = 0xFF;
+	mode = 0xFF;
+	//LEDs
+	DDRA |= (1<<LED_PWR);
+
+	//pullup
+	PORTD |= (1<<FT_CBUS1);
+
+	uart_init(UART_BAUD_SELECT(UART_BAUD_RATE, F_CPU));
+
+	//init SPI
+	DDRB |= (1<<PB5) | (1<<PB7)| (1<<DDB4);
+
+	// SPI Type: Master
+	// SPI Clock Rate: 62,500 kHz
+	// SPI Clock Phase: Cycle Start
+	// SPI Clock Polarity: Low
+	// SPI Data Order: MSB First
+	SPCR=0x57;//at 8MHz
+	//SPCR=0x56;//at 4MHz
+	SPSR=0x00;
+
+	//beta commands -> cdc
+	TCCR1B |= (1<<CS11);    // no prescaler 8 -> 1 timer clock tick is 1us long
```

Comment:
[`TCCR1B`](http://www.avrbeginners.net/architecture/timers/timers.html) is an AVR timer. `CS11` is a "clock select" bit, which basically changes the clock rate on the board, with the definition of:

 - Divide Clock by 8
 - Divide Clock by 64
 - Increment Timer 1 on T1 Pin Falling Edge
 - Increment Timer 1 on T1 Pin Rising Edge

`TCNT1` counts system clock ticks according to the frequency specified.